scripts/set_icmp_state

```diff
@@ -0,0 +1,293 @@
+
+There are two scopes: you can block/allow traffic to all destinations, or block/allow traffic
+to specific destinations. The "all" rule will be be given higher priority (9999) than rules to
+specific destinations (9995). SONiC supports inserting multiple rules at the same priority, so
+all destination-specific rules will be given 9995.
+
+You can also specify a VLAN ID to restrict the block to a given VLAN destination. Note that you
+cannot have a rule that blocks all and blocks all on a certain VLAN in the same table (e.g.
+if you have blocked all traffic on table A vlan 100, you cannot also block all traffic on table A with
+no VLAN ID).
+
+Example:
+
+Block/unblock all ICMP traffic to Vlan101 on V4_VLAN_101_OUT:
+./set_icmp_state block all V4_VLAN_101_OUT -v 101
+./set_icmp_state allow all V4_VLAN_101_OUT
+
+Block/unblock all ICMP traffic to 10.0.0.37/32 on Vlan101 on V4_VLAN_101_OUT:
+./set_icmp_state block destination V4_VLAN_101_OUT 10.0.0.37/32 -v 101
+./set_icmp_state allow destination V4_VLAN_101_OUT 10.0.0.37/32
+"""
+
+from __future__ import print_function
+
+import syslog
+import sys
+import click
+import ipaddress
+
+from swsssdk import ConfigDBConnector
+
+CONFIG_DB_ACL_TABLE_TABLE = "ACL_TABLE"
+CONFIG_DB_ACL_RULE_TABLE = "ACL_RULE"
+CONFIG_DB_VLAN_TABLE = "VLAN"
+
+BLOCK_ALL_KEY = "RULE_BLOCK_ALL_ICMP"
+BLOCK_KEY_PREFIX = "RULE_BLOCK_ICMP_"
+
+IP_VERSION_V4 = "V4"
+IP_VERSION_V6 = "V6"
+
+ICMP_IP_PROTOCOL_NUM = "1"
+ICMPV6_NEXT_HEADER_NUM = "58"
+
+BLOCK_ALL_PRIORITY = "9999"
+BLOCK_DEST_PRIORITY = "9995"
+
+
+def notice(msg):
+    """Log a NOTICE message to the console and syslog."""
+    syslog.syslog(syslog.LOG_NOTICE, msg)
+    print(msg)
+
+
+def error(msg):
+    """Log an ERR message to the console and syslog, and exit the program with an error code."""
+    syslog.syslog(syslog.LOG_ERR, msg)
+    print(msg, file=sys.stderr)
+    sys.exit(1)
+
+
+def verify_table_exists(configdb, table_name):
+    """Verify that a given ACL table exists in Config DB."""
+    target_table = configdb.get_entry(CONFIG_DB_ACL_TABLE_TABLE, table_name)
+
+    if not target_table:
+        error("Table {} not found, exiting...".format(table_name))
+
+    return target_table
+
+
+def verify_valid_ip_address(ip_address, table_ip_version):
+    """Verify that a given IP address is valid.
+
+    This checks that 1) the provided address is valid, and 2) that the IP version of the
+    address matches the IP version of the table this address will be applied to.
+    """
+    try:
+        destination_ip_version = ipaddress.ip_network(ip_address).version
+        table_ip_version = 6 if table_ip_version == IP_VERSION_V6 else 4
+        if destination_ip_version != table_ip_version:
+            error("Cannot use an IPv{} rule with an IPv{} table, exiting...".format(destination_ip_version, table_ip_version))
+    except Exception as e:
+        error("Could not parse {} as a valid IP address; exception={}".format(ip_address, e))
```

Comment:
>exception [](start = 57, length = 9)

also print its stack?

Reply:
`exception={}` should cover that, I think? Are you looking for something more specific?